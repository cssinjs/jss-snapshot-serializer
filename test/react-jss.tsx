import React from 'react';
import { createUseStyles } from 'react-jss';
import { render } from '@testing-library/react';
import serializer from '../src/jss-snapshot-serializer';

expect.addSnapshotSerializer(serializer);

const useStyles = createUseStyles({
  redBox: {
    width: 100,
    height: 100,
    backgroundColor: 'red',
  },
  blueBox: {
    width: 100,
    height: 100,
    backgroundColor: 'blue',
  },
});

const RedBox = () => {
  const classes = useStyles();
  return <div className={classes.redBox} />;
};

const BlueAndRed = () => {
  const classes = useStyles();
  return (
    <div>
      <div className={classes.redBox} />
      <div className={classes.blueBox} />
    </div>
  );
};

const BlueAndRedNesting = () => {
  const classes = useStyles();
  return (
    <div>
      <div className={classes.redBox} />
      <div>
        <>
          <div className={classes.redBox} />
          <div className={classes.blueBox} />
        </>
        <div className={classes.blueBox} />
      </div>
    </div>
  );
};

describe('react-jss', () => {
  it('removes non-deterministic class names', () => {
    const redBox = render(<RedBox />).asFragment();
    expect(redBox).toMatchSnapshot();
  });

  it('can handle components with several styles', () => {
    const blueRed = render(<BlueAndRed />).asFragment();
    expect(blueRed).toMatchSnapshot();
  });

  it('can handle components with nesting', () => {
    const blueRed = render(<BlueAndRedNesting />).asFragment();
    expect(blueRed).toMatchSnapshot();
  });
});
