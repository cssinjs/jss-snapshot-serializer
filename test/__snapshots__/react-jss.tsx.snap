// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-jss can handle components with nesting 1`] = `
<DocumentFragment>
  .j0 {
    width: 100px;
    height: 100px;
    background-color: red;
  }

  .j1 {
    width: 100px;
    height: 100px;
    background-color: blue;
  }

  <div>
    <div
      class="j0"
    />
    <div>
      <div
        class="j0"
      />
      <div
        class="j1"
      />
      <div
        class="j1"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`react-jss can handle components with several styles 1`] = `
<DocumentFragment>
  .j0 {
    width: 100px;
    height: 100px;
    background-color: red;
  }

  .j1 {
    width: 100px;
    height: 100px;
    background-color: blue;
  }

  <div>
    <div
      class="j0"
    />
    <div
      class="j1"
    />
  </div>
</DocumentFragment>
`;

exports[`react-jss removes non-deterministic class names 1`] = `
<DocumentFragment>
  .j0 {
    width: 100px;
    height: 100px;
    background-color: red;
  }

  <div
    class="j0"
  />
</DocumentFragment>
`;
